pkg/storage/leveldb/leveldb_store.go

```diff
@@ -8,45 +8,100 @@ package leveldb
 
 import (
 	"errors"
+	"fmt"
 	"strings"
+	"sync"
 
 	"github.com/syndtr/goleveldb/leveldb"
 
 	"github.com/hyperledger/aries-framework-go/pkg/storage"
 )
 
+const pathPattern = "%s-%s"
+
 // Provider leveldb implementation of storage.Provider interface
 type Provider struct {
-	db *leveldb.DB
+	dbPath string
+	dbs    map[string]*leveldbStore
+	lock   sync.RWMutex
 }
 
 // NewProvider instantiates Provider
 func NewProvider(dbPath string) (*Provider, error) {
-	db, err := leveldb.OpenFile(dbPath, nil)
+	return &Provider{dbs: make(map[string]*leveldbStore), dbPath: dbPath}, nil
+}
+
+// OpenStore opens and returns a store for given name space.
+func (p *Provider) OpenStore(name string) (storage.Store, error) {
+	store := p.getLeveldbStore(name)
+	if store == nil {
+		return p.newLeveldbStore(name)
+	}
+	return store, nil
+}
+
+// getLeveldbStore finds level db store with given name
+// returns nil if not found
+func (p *Provider) getLeveldbStore(name string) *leveldbStore {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+	return p.dbs[strings.ToLower(name)]
+}
+
+// newLeveldbStore creates level db store for given name space
+// returns nil if not found
+func (p *Provider) newLeveldbStore(name string) (*leveldbStore, error) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	db, err := leveldb.OpenFile(fmt.Sprintf(pathPattern, p.dbPath, name), nil)
 	if err != nil {
 		return nil, err
 	}
-	return &Provider{db}, nil
+
+	store := &leveldbStore{db}
+	p.dbs[strings.ToLower(name)] = store
+	return store, nil
 }
 
-// GetStoreHandle returns a handle to the store
-func (provider *Provider) GetStoreHandle() (storage.Store, error) {
-	return newLeveldbStore(provider.db), nil
+// CloseAll closes the underlying provider
+func (p *Provider) CloseAll() error {
```

Comment:
`Close` is probably a sufficient name as you are closing the store provider.

Reply:
ok, will rename it to close

Reply:
done